### DISCUSSION TITLE
Precompiling Assets When Building In Rails

### DISCUSSION THREAD

**Discussion by titan2gman** [titan2gman/designbro-v2#1] [Q&A] [answered]
I'm using a Dockerfile as a starting point for a Rails 5 project. I'm running into problems when the build gets to the assets:precompile step—it's trying to connect to the database and fails. I tried setting config.assets.initialize_on_precompile = false, but that didn't resolve the issue.

When the database host is missing, or set to "localhost," Rails tries connecting to the database via socket. If I add the host as part of the precompile command (MYSQL_HOST=db SECRET_KEY_BASE=dummyvalue rails assets:precompile), then it doesn't like the host ("Unknown MySQL server host 'db').

Have you ever run into this issue on your projects using this Dockerfile?

**Comment by fortune-web** [comment, marked as answer]
Hi,

I vaguely remember this being an issue with Rails 5.

I'm almost sure I was able to set DATABASE_URL=postgresql://dummy:dummy@127.0.0.1/dummy SECRET_KEY_BASE=dummy rails assets:precompile and it worked. In your case try swapping postgresql with mysql. I remember having to set the whole DB URI for it to work.

>I actually think that the "strict effects" thing is going to be a much bigger breaking change in the ecosystem than Concurrency:
>
>const didLogRef = useRef(false);
>
>useEffect(() => {
>  // In this case, whether we are mounting or remounting,
>  // we use a ref so that we only log an impression once.
>  if (didLogRef.current === false) {
>    didLogRef.current = true;
>
>    SomeTrackingAPI.logImpression();
>  }
>}, []);
>We've all been trained for the last two years that useEffect(cb, []) is how we mimic cDM (yes, yes, I know "they're not the same thing!"), and how to fetch once on render.
>
>if I'm reading these pages right:
>
>In prod, this would only run more than once if the component gets Suspense-hidden or similar
>But it's going to double-run inside of <StrictMode>
>And all the CRA templates and such come with <StrictMode> on by default
>So, I foresee a lot of people getting confused at "why did my API call run twice?" here.
>
>I almost see a need for an official useRunThisOnlyOnceOnMountNoKiddingIMeanIt() hook recipe that can be copy-pasted to do the ref+effect combination above
>
>Also, as a side note, this initial code snippet seems odd:
>
>```
const getterRef = useRef(() => {
>  if (ref.current === null) {
>    ref.current = new SomeImperativeThing();
>  }
>  return ref.current;
>});
```
>I was pretty sure that useRef doesn't take a callback per se, at least not in the "run this code" sense - it just does const ref = {current: initialValue}; internally ( per this line in ReactFiberHooks.js).
>
>I think the hooks docs show doing that lazy init while rendering specifically, as the rare edge cases it's legal to mutate while rendering:
>
>https://reactjs.org/docs/hooks-faq.html#how-to-create-expensive-objects-lazily
>
>I assume that's the intended usage pattern here rather than a callback.
>
>edit
>
>Actually, maybe I misread what the "lazy init function" is here. Is it supposed to be something that's manually called elsewhere? I think I can sorta see how that pattern would work, just not something I'm used to seeing.
